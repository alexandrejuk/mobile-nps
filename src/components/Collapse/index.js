import React, {
  useRef,
  useState,
} from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import arrowIcon from '../../assets/icons/arrow-up.svg'
import Title from '../Title'
import styles from './style.module.css'

const calcHeight = value => {
  if (value === 57) {
    return 147
  }
  return (value + 56)
}

const Collapse = ({
  children,
  title,
}) => {
  const [collapsed, setCollapsed] = useState(false)
  const [height, setHeight] = useState(65)
  const ref = useRef(null)

  const openCollapse = () => {
    let newHeight = 65
    const { current: { clientHeight } } = ref
    if (!collapsed) {
      newHeight = calcHeight(clientHeight + 50)
    }
    setHeight(newHeight)
    return setCollapsed(!collapsed)
  }

  return (
    <div
      className={styles.collapse}
      style={{ height: `${height}px` }}
    >
      <div
        className={styles.collapseHeader}
        onKeyPress={openCollapse}
        onClick={openCollapse}
        tabIndex="0"
        role="button"
      >
        <Title
          size="medium"
          text={title}
        />
        <div className={classNames(
          styles.icon, {
            [styles.collapseIcon]: collapsed,
          }
        )}>
          <img src={arrowIcon} alt="arrow" />
        </div>
      </div>
      <div
        className={styles.collapseBody}
        ref={ref}
      >
        { children }
      </div>
    </div>
  )
}

Collapse.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
}

export default Collapse
