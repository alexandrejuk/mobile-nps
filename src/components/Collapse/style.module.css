@import '../../assets/css/colors.css';

.collapse {
  background-color: var(--color-white);
  border: solid 1px var( --color-alto);
  border-radius: 3px;
  box-shadow: 0px 0px 16px rgba(0, 0, 0, 0.15);
  position: relative;
  overflow: hidden;
  transition: height 0.4s ease-in-out;
}

.collapseHeader {
  align-items: center;
  justify-content: space-between;
  display: flex;
  cursor: pointer;
  border-bottom: solid 1px var( --color-alto);
  padding: 10px 20px;
}

.collapseHeader:focus {
  outline: none;
}

.collapseBody {
  font-family: 'Rubik', Arial, Helvetica, sans-serif;
  padding: 5px 20px;
  margin-top: 16px;
}

.icon {
  transform: rotate(-180deg);
  transition: transform 0.4s ease-in-out;
}

.collapseIcon {
  transform: rotate(-360deg);
}
