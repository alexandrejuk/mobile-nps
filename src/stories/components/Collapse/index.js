import React from 'react'
import { Collapse } from '../../../components'

export default {
  title: 'Components/Collapse',
}

export const Default = () => (
  <Collapse title="Collapse title">
    <p>
      Lorem ipsum color, danger baba
      Lorem ipsum color, danger baba
      Lorem ipsum color, danger baba
      Lorem ipsum color, danger baba
      Lorem ipsum color, danger baba
      Lorem ipsum color, danger baba
      Lorem ipsum color, danger baba
    </p>
  </Collapse>
)
