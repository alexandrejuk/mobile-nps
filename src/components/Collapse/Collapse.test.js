import React from 'react'
import { render } from '@testing-library/react'

import Collapse from './'

describe('Collapse component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Collapse
        title="I am collapse component"
      >
        Lorem ipsum color, danger baba
      </Collapse>
    )
    expect(container).toMatchSnapshot()
  })
})
