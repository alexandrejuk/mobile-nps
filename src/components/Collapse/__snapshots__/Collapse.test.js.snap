// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse component should render correctly 1`] = `
<div>
  <div
    class="collapse"
    style="height: 65px;"
  >
    <div
      class="collapseHeader"
      role="button"
      tabindex="0"
    >
      <h1
        class="primary medium left title normal"
      >
        I am collapse component
      </h1>
      <div
        class="icon"
      >
        <img
          alt="arrow"
          src="arrow-up.svg"
        />
      </div>
    </div>
    <div
      class="collapseBody"
    >
      Lorem ipsum color, danger baba
    </div>
  </div>
</div>
`;
